templates/common/_base/files/configure-ovs-network.yaml

```diff
@@ -94,12 +94,33 @@ contents:
         nmcli c add type ovs-port conn.interface br-ex master br-ex con-name ovs-port-br-ex
       fi
 
+      extra_phys_args=""
+      # check if this interface is a vlan, bond, or ethernet type
+      if nmcli --fields connection.type conn show ${old_conn} | grep -q vlan; then
+        iface_type=vlan
+        vlan_id=$(nmcli --fields vlan.id conn show ${old_conn} | awk '{print $2}')
+        if [ -z "$vlan_id"]; then
+          echo "ERROR: unable to determine vlan_id for vlan connection: ${old_conn}"
+          exit 1
+        fi
+        vlan_parent=$(nmcli --fields vlan.parent conn show ${old_conn} | awk '{print $2}')
+        if [ -z "$vlan_parent" ]; then
+          echo "ERROR: unable to determine vlan_parent for vlan connection: ${old_conn}"
+          exit 1
+        fi
+        extra_phys_args="dev ${vlan_parent} id ${vlan_id}"
+      elif nmcli --fields connection.type conn show ${old_conn} | grep -q bond || [ -d /sys/class/net/${iface}/bonding/ ]; then
+        iface_type=bond
+      else
+        iface_type=802-3-ethernet
```

Comment:
should we explicitly validate that and properly error out if it's any other type?

Reply:
I'm kind of hesitant to do that. If we don't explicitly know the type I think its ok to attempt with ethernet. If it doesn't work we script will end up failing and fall back to the original connection.